posthog/api/organization_member.py

```diff
@@ -38,6 +39,38 @@ class Meta:
         model = OrganizationMembership
         fields = ["membership_id", "user_id", "user_first_name", "user_email", "level", "joined_at", "updated_at"]
 
+    def update(self, updated_membership, validated_data):
+        updated_membership = cast(OrganizationMembership, updated_membership)
+        raise_errors_on_nested_writes("update", self, validated_data)
+        requesting_membership: OrganizationMembership = OrganizationMembership.objects.get(
+            organization=updated_membership.organization, user=self.context["request"].user
+        )
+        new_level = validated_data.get("level")
+        with transaction.atomic():
+            if new_level is not None:
```

Comment:
Suggestion: Extract all the validation to separate method called `validate_level_change`, test fully.